scripts/TestGen.hx

```diff
@@ -0,0 +1,380 @@
+/*
+
+This test generator was initially written by David Bruce: https://github.com/davidbruce
+
+His initial PR was stale and his commits were accidentally nuked, hence this attribution.
+
+*/
+
+import haxe.Template;
+import haxe.Json;
+import sys.io.File;
+import sys.FileSystem;
+import sys.Http;
+
+using StringTools;
+
+class TestGen
+{
+	static var exercise:String;
+	static var exerciseStubTmpl = new Template("package;
+
+class ::exercise:: {
+    ::methods:: 
+}");
+
+	static var methodsTmpl = new Template("public static function ::method::(::args::): ::returnType:: {
+        throw \"Not Implemented\"; // Delete this statement and write your own implementation.
+    }");
+
+	static var testTmpl = new Template("package;
+
+using buddy.Should;
+
+//Created by testgen.hx
+class Test extends buddy.SingleSuite {
+    public function new() {
+        ::stories::
+    }
+}");
+
+	static var storyTmpl = new Template("describe(\"::story::\", {
+        ::testCases::});");
+
+	static var testCaseTmpl = new Template("    it(\"::testCase::\", {
+                ::testCriteria::
+            });
+        ");
+
+	public static function main()
+	{
+		// TODO: Configlet is able to create the directories/files we need:
```

Comment:
If we do run configlet to build the structure, the new exercise already needs to be listed in the track config.json within the `practice` array under `exercises`. If it isn't, configlet won't add the files so we should alert the user and bail.